usrsctplib/netinet/sctputil.c

```diff
@@ -6667,7 +6667,7 @@ sctp_sorecvmsg(struct socket *so,
 					copied_so_far += cp_len;
 					freed_so_far += (uint32_t)cp_len;
 					freed_so_far += MSIZE;
-					atomic_subtract_int(&control->length, cp_len);
+					atomic_subtract_int(&control->length, (int32_t)cp_len);
```

Comment:
Fixed in https://github.com/sctplab/usrsctp/commit/078bd24ac1768bc6dc87d40b030434cba1987ea3.